jerry-core/ecma/builtin-objects/ecma-builtin-string-prototype.cpp

```diff
@@ -286,7 +286,48 @@ static ecma_completion_value_t
 ecma_builtin_string_prototype_object_locale_compare (ecma_value_t this_arg, /**< this argument */
                                                      ecma_value_t arg) /**< routine's argument */
 {
-  ECMA_BUILTIN_CP_UNIMPLEMENTED (this_arg, arg);
+  ecma_completion_value_t ret_value = ecma_make_empty_completion_value ();
+
+  /* 1. */
+  ECMA_TRY_CATCH (this_check_coercible_val,
+                  ecma_op_check_object_coercible (this_arg),
+                  ret_value);
+
+  /* 2. */
+  ECMA_TRY_CATCH (this_to_string_val,
+                  ecma_op_to_string (this_arg),
+                  ret_value);
+
+  /* 3. */
+  ECMA_TRY_CATCH (arg_to_string_val,
+                  ecma_op_to_string (arg),
+                  ret_value);
+
+  ecma_string_t *this_string_p = ecma_get_string_from_completion_value (this_to_string_val);
+  ecma_string_t *arg_string_p = ecma_get_string_from_completion_value (arg_to_string_val);
```

Comment:
The `this_to_string_val` and `arg_to_string_val` are not completion values. They are `ecma_value_t`-s